### PR TITLE
chore: Modernise Android build config for plugin and example

### PR DESCRIPTION
**Pull request by vbuberen**
While going through the source code in the repo I saw that plugin and example app both have quiet outdated Android build configuration and decided to help bring it up to date a little bit.

- Updated used Android Gradle plugin to `7.2.0`  in both plugin and example, which is latest at the moment. Needed to be able to build project in latest Android Studio versions as `3.2` is already minimal supported version: https://developer.android.com/studio/releases/gradle-plugin
- Updated Gradle wrapper version to the latest `7.4.2` in both plugin and example as after changing version Android Gradle version we had to update wrapper anyway.
- Changed `compileSDK` to 31 for plugin and `targetSDK` for example to compile against currently latest Android SDK.
- Added `exported=true` in AndroidManifest of example as it is a requirement in SDK 31.
- Changed AndroidManifest in example to declare app name as `android:name="${applicationName}"` to not have build issues saying that `example` uses Flutter embedding V1.
- Added `android.useAndroidX=true` flag as project has dependencies which use AndroidX, but doesn't enable AndroidX usage itself, so you can see following warnings during build:
androidxIssues

P.S. `pubspec.lock` versions changed because I was building the project after my changes and had to do `pub.get`. Version bump for Dart happened because I use latest stable versions of Dart and Flutter. These updates would need to happen anyway, so didn't revert them.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,9 @@
 ios/Flutter/flutter_export_environment.sh
 .idea/**
 
+# MacOS generated file
+.DS_Store
```
Comment (vbuberen) [inline .gitignore:6]: Removed such file from repo and added to `gitignore` as it is not something that has to be in the repo.

**Review by danielgomezrico — APPROVED** [review]
Thanks for this 👍🏻

**Comment by vbuberen** [comment]
`Build Android` job fails due to Github Actions using Flutter 2.0.0, which couldn't work with Gradle 7 and newer. 

Here is the issue in the Flutter repo with fix, which adds the compatibility: https://github.com/flutter/flutter/pull/79167

Thus, will update version of Flutter used on CI in this PR as well to make all actions pass.

**Comment by danielgomezrico** [comment]
@vbuberen what are the consequences of updating the library to flutter 3.0?

**Comment by vbuberen** [comment]
In the context of this PR nothing changes for the project because of building with Dart 2.17 and Flutter 3 as I introduced no new code or features, which appeared only in mentioned releases. 

In case you would like to check yourself here are links with release notes descriptions:
https://docs.flutter.dev/development/tools/sdk/release-notes/release-notes-3.0.0
https://medium.com/dartlang/dart-2-17-b216bfc80c5d

**Comment by vbuberen** [comment]
@danielgomezrico Do you you want me to address something else in this PR or it can be merged? 
Would like to open one more PR with some Segment Android SDK updates, but feel like we might want to merge this one first.

**Review by danielgomezrico — APPROVED** [review]
LGTM

**Comment by vbuberen** [comment]
Thanks for approval. I was curios more on when this PR will be merged 😄

**Comment by danielgomezrico** [comment]
@vbuberen LGTM thanks for opening the PR!

**Pull request merged**